test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs

```diff
@@ -255,19 +198,50 @@ public void CreateArgumentShouldAddMultipleTestAdapterPaths()
         [TestMethod]
         public void CreateArgumentShouldAddMultipleLoggers()
         {
-            var vstestTask = new VSTestTask();
+            this.vsTestTask.VSTestLogger = new string[] { "trx;LogFileName=foo bar.trx", "console" };
+            var allArguments = this.vsTestTask.CreateArgument().ToArray();
 
-            // Add values for required properties.
-            vstestTask.TestFileFullPath = "abc";
-            vstestTask.VSTestFramework = "abc";
+            Assert.IsNotNull(allArguments.FirstOrDefault(arg => arg.Contains("--logger:\"trx;LogFileName=foo bar.trx\"")));
+            Assert.IsNotNull(allArguments.FirstOrDefault(arg => arg.Contains("--logger:console")));
+        }
 
-            vstestTask.VSTestLogger = new string[] { "trx;LogFileName=foo bar.trx", "console" };
+        [TestMethod]
+        public void CreateArgumentShouldAddTraceCollectorDirectoryPathAsTestAdapterForCodeCoverageCollect()
+        {
+            const string traceDataCollectorDirectoryPath = @"c:\path\to\tracedata collector";
+            this.vsTestTask.VSTestTraceDataCollectorDirectoryPath = traceDataCollectorDirectoryPath;
+            this.vsTestTask.VSTestCollect = new string[] { "code coverage" };
 
+            var allArguments = this.vsTestTask.CreateArgument().ToArray();
 
-            var allArguments = vstestTask.CreateArgument().ToArray();
+            const string expectedArg = "--testAdapterPath:\"c:\\path\\to\\tracedata collector\"";
+            CollectionAssert.Contains(allArguments, expectedArg, $"Expected argument: '''{expectedArg}''' not present in [{string.Join(", ", allArguments)}]");
+        }
 
-            Assert.IsNotNull(allArguments.FirstOrDefault(arg => arg.Contains("--logger:\"trx;LogFileName=foo bar.trx\"")));
-            Assert.IsNotNull(allArguments.FirstOrDefault(arg => arg.Contains("--logger:console")));
+        [TestMethod]
+        public void CreateArgumentShouldNotAddTraceCollectorDirectoryPathAsTestAdapterForNonCodeCoverageCollect()
+        {
+            const string traceDataCollectorDirectoryPath = @"c:\path\to\tracedata collector";
+            this.vsTestTask.VSTestTraceDataCollectorDirectoryPath = traceDataCollectorDirectoryPath;
+            this.vsTestTask.VSTestCollect = new string[] { "not code coverage" };
+
+            var allArguments = this.vsTestTask.CreateArgument().ToArray();
+
+            const string notExpectedArg = "--testAdapterPath:\"c:\\path\\to\\tracedata collector\"";
+            CollectionAssert.DoesNotContain(allArguments, notExpectedArg, $"Not expected argument: '''{notExpectedArg}''' present in [{string.Join(", ", allArguments)}]");
+        }
+
+        [TestMethod]
+        public void CreateArgumentShouldAddTraceCollectorDirectoryPathAsTestAdapterIfSettingsGiven()
```

Comment:
Nit: Check if we can add another test where VSTestTraceDataCollectorDirectoryPath is empty, so it is not set if if runsettings is present

Reply:
Done.